examples/web/cache/refresh-disposable-tokens.ts

```diff
@@ -0,0 +1,238 @@
+import {
+  AuthClient,
+  CredentialProvider,
+  DisposableTokenScope,
+  ExpiresAt,
+  ExpiresIn,
+  GenerateDisposableTokenResponse,
+  TopicClient,
+  TopicItem,
+  TopicPermission,
+  TopicPublishResponse,
+  TopicRole,
+  TopicSubscribe,
+  TopicSubscribeResponse,
+} from '@gomomento/sdk-web';
+import {initJSDom} from './utils/jsdom';
+
+function getRefreshAfterMs(expiresAt: ExpiresAt, refreshBefore: number): number {
+  const refreshingIn = expiresAt.epoch() * 1000 - Date.now() - refreshBefore;
+  console.log(`Refreshing in ${refreshingIn} ms`);
+  return refreshingIn;
+}
+
+// In your own setup, the token vending machine would likely be a separate service.
+// See https://github.com/momentohq/client-sdk-javascript/tree/main/examples/nodejs/token-vending-machine
+async function tokenVendingMachine(): Promise<{token: string; expiresAt: ExpiresAt}> {
+  const authClient = new AuthClient({});
```

Comment:
Yes, that's used only in the `localTokenVendingMachine` example for anyone who wants to run the example completely locally (in case they don't have a deployed TVM). Then they can just run this example like any other one, `export MOMENTO_API_KEY=<api key> npm run tokens`.

If they want to run the example against their own deployed TVM, then they can pass in an endpoint instead and switch to using the `tokenVendingMachine` function in the example code. Running the example that way would look like: `TVM_ENDPOINT=<their TVM url> npm run tokens`

Reply:
@anitarua 
So `localTokenVendingMachine` is something they will integrate into their backend so that api can be called from each iPad? 
But where is `localTokenVendingMachine` using `MOMENTO_API_KEY`?

Reply:
Yes, their backend would call `generateDisposableToken` and return that token to the iPads (or other end user devices). 

`MOMENTO_API_KEY` is used to create an `AuthClient`. This auth client in the example is using default configuration settings: `CredentialProvider` that expects a `MOMENTO_API_KEY` environment variable and default `AuthClientConfiguration`.

Their backend component that creates the `AuthClient` and calls `generateDisposableToken` could be a Lambda function, like the TVM example is, or it can be something they write into their own server.